rebac-admin-backend/v1/error.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2024 Canonical Ltd.
+
+package v1
+
+// UnauthorizedError represents unauthorized access error.
+type UnauthorizedError struct{}
```

Comment:
suggestion
```
var (
   UnauthorizedError = errors.New("unauthorized")
   NotFoundError = errors.New("not found")
)
```

Reply:
I thought of that, but we might want to add more information to the errors down the road. For example, in case of `NotFoundError`, for some endpoints there are more than one entity that could be missing. For example, for `PATCH /groups/{id}/identities`, the group ID or any of the identity IDs could be missing. So, for more descriptive errors we might want to have such information attached.

I haven't yet added fields to the error types, because we need to figure it out yet; either we define some fields on them, or we only change the string message. In any way, we'd need a separate error struct type.